llama-index-integrations/vector_stores/llama-index-vector-stores-falkordb/llamaindex/falkordbvector/base.py

```diff
@@ -0,0 +1,216 @@
+            text_node_property=text_node_property,
+            embedding_dimension=embedding_dimension,
+        )
+
+        if distance_strategy not in ["cosine", "euclidean"]:
+            raise ValueError("distance_strategy must be either 'euclidean' or 'cosine'")
+
+        self._driver = FalkorDB.from_url(url).select_graph(database)
+        self._database = database
+
+        check_if_not_null(
+            [
+                "index_name",
+                "node_label",
+                "embedding_node_property",
+                "text_node_property",
+            ],
+            [index_name, node_label, embedding_node_property, text_node_property],
+        )
+
+        if not self.retrieve_existing_index():
+            self.create_new_index()
+
+    @property
+    def client(self) -> FalkorDB:
+        return self._driver
+
+    def create_new_index(self) -> None:
+        index_query = (
+            f"CREATE VECTOR INDEX {self.index_name} "
+            f"FOR (n:`{self.node_label}`) "
+            f"ON (n.`{self.embedding_node_property}`) "
+            f"OPTIONS {{dimension: {self.embedding_dimension}, metric: '{self.distance_strategy}'}}"
+        )
+        self._driver.query(index_query)
+
+    def retrieve_existing_index(self) -> bool:
+        index_information = self._driver.query(
+            "SHOW INDEXES "
+            "YIELD name, type, entityType, properties, options "
+            "WHERE type = 'VECTOR' AND name = $index_name",
+            params={"index_name": self.index_name}
+        )
+        if index_information.result_set:
+            index = index_information.result_set[0]
+            self.node_label = index['entityType']
+            self.embedding_node_property = index['properties'][0]
+            self.embedding_dimension = index['options']['dimension']
+            return True
+        return False
+
+    def add(self, nodes: List[BaseNode], **add_kwargs: Any) -> List[str]:
+        ids = [r.node_id for r in nodes]
+        import_query = (
+            "UNWIND $data AS row "
+            f"MERGE (c:`{self.node_label}` {{id: row.id}}) "
+            f"SET c.`{self.embedding_node_property}` = row.embedding, "
+            f"c.`{self.text_node_property}` = row.text, "
+            "c += row.metadata"
+        )
+
+        self._driver.query(import_query, params={"data": clean_params(nodes)})
+        return ids
+
+    def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResult:
+        base_query = (
+            f"MATCH (n:`{self.node_label}`) "
+            f"WHERE n.`{self.embedding_node_property}` IS NOT NULL "
+        )
+
+        if query.filters:
+            filter_snippets, filter_params = construct_metadata_filter(query.filters)
+            base_query += f"AND {filter_snippets} "
+        else:
+            filter_params = {}
+
+        similarity_query = (
+            f"WITH n, vector.similarity.{self.distance_strategy}("
```

Comment:
see https://docs.falkordb.com/cypher/indexing.html#vector-indexing how to use vector indexing